nexus/reconfigurator/simulation/src/system.rs

```diff
@@ -135,16 +135,52 @@ impl SimSystem {
         self.collections.values().map(|c| &**c)
     }
 
+    pub fn resolve_blueprint_id(
+        &self,
+        original: BlueprintId,
+    ) -> Result<ResolvedBlueprintId, KeyError> {
+        let resolved = match original {
+            BlueprintId::Target => {
+                self.target_blueprint
+                    .ok_or_else(|| KeyError::blueprint(original))?
+                    .target_id
+            }
+            BlueprintId::Latest => {
+                // self.blueprints is an IndexMap, so the last element is the
+                // latest.
```

Comment:
Is this true even if the system state has been loaded from a serialized JSON blob?

Reply:
Hmm, good q, probably not...

I guess I should fix this and also add a test for it.

Reply:
Or, I guess, have an explicit data structure that combines a hashmap and a btreemap. That's probably easier to reason about if nothing else.

Reply:
Not too bad in the end -- added an `insert_sorted_by` method which helps enforce the invariant.